src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java

```diff
@@ -175,6 +175,19 @@ public static void scan(Run build, PrintStream buildLog) {
                 printDownstream(buildLog, downstreamFailureCauses);
             }
 
+            if (PluginImpl.getInstance().isEnableBuildDescription() && !foundCauseList.isEmpty()) {
+              // create a build description text from the list of failures.
+              String buildDescription = "<mark><b>";
+              if (foundCauseList.get(0).getCategories() != null) {
+                buildDescription = buildDescription.concat(foundCauseList.get(0).getCategories().get(0));
+                buildDescription = buildDescription.concat(": ");
+              }
+              buildDescription = buildDescription.concat("</b> <i>");
+              buildDescription = buildDescription.concat(foundCauseList.get(0).getDescription());
+              buildDescription = buildDescription.concat("</i></mark>");
+              build.setDescription(buildDescription);
```

Comment:
There is also the build description plugin that writes the description in a post build step, as well as pipeline that could write a description whenever it wants during the build. So better to append than overwrite.